gcovr/tests/test_gcovr.py

```diff
@@ -232,15 +240,20 @@ def test_build(compiled, format, available_targets):
 
     for coverage_file, reference_file in find_reference_files(output_pattern):
         with io.open(coverage_file, encoding=encoding) as f:
-            coverage = scrub(f.read())
+            coverage_raw = f.read()
+            coverage = scrub(coverage_raw)
         with io.open(reference_file, encoding=encoding) as f:
             reference = scrub(f.read())
 
         if assert_equals is not None:
             assert_equals(coverage, reference)
         else:
-            assert coverage == reference, "coverage={}, reference={}".format(
-                coverage_file, reference_file)
+            diff_out = list(difflib.unified_diff(reference.splitlines(keepends=True), coverage.splitlines(keepends=True), fromfile=reference_file, tofile=coverage_file))
+            len_diff_is_zero = True if len(diff_out) == 0 else False
```

Comment:
Explicitly listing true/false seems more complicated than just using the condition. Consider:

```suggestion
            diff_is_empty = len(diff_out) == 0
```

Reply:
I'll change this and add an option '--generate-reference' which will only generate non existing references.

Reply:
The option `--generate-reference` only creates non existing references.
The option `--update-reference` updates all existing references if there are changes in the normalized files.